.github/workflows/tests_scripts.yml

```diff
@@ -99,6 +101,20 @@ jobs:
           # check if tarballs have been produced
           ls -l *.tar.gz
 
+    - name: test sign_verify_file_ssh.sh script
+      run: |
+          # Create a PEM format ssh identity
+          ssh-keygen -t rsa -b 4096 -m PEM -f id_rsa.pem -N ""
+          # Find a file to sign
+          export TARBALL="$(ls *.tar.gz | tail -1)"
+          # Sign the file
+          ./sign_verify_file_ssh.sh sign "$TARBALL" id_rsa.pem
+          # Create an allowed_signers file based on the public key
+          echo -n "allowed_signer " > allowed_signers 
+          cat id_rsa.pem.pub >> allowed_signers
+          # Verify the signature
+          ./sign_verify_file_ssh.sh verify "$TARBALL" allowed_signers
+
```

Comment:
A bit unsure what this **actually** tests? To me it looks like if the script exists in the current directory, is executable and exits with 0 the test will succeed?

Reply:
This tests the workflow end-to-end, it creates a key (PEM format, same as a GitHub App key), signs a file with it, and then verifies the signature.

Basically it is testing core functionality, if there is a problem, the code will error out and the CI will fail.

Reply:
The test succeeds if the last run of `sign_verify_file_ssh.sh` exits with `0`, no? All the rest that is done in the test doesn't really matter. Wouldn't be better to use predefined input (data file, pem key) & output (pub key, signature) instead of generating them on the fly? Thus you would have input and output under control and could verify if the created pub key and signature match what you expect.

Reply:
Not quite, the test will also fail if any of the previous commands fail. The first command tests the signing, the second command tests the verification. The specific data file and PEM key are not particularly relevant (we're not processing the data itself, only the signature file that gets created)

Reply:
Maybe call it a unit test :shrug: